File: C++/2-Medium/00128._最长连续序列/00128._最长连续序列.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

// 函数：计算最长连续序列的长度
int longestConsecutive(vector<int>& nums) {
    unordered_set<int> num_set(nums.begin(), nums.end());
    int longest = 0;

    for (int num : num_set) {
        // 只有当当前数字是连续序列的起始点时，才开始计算
        if (num_set.find(num - 1) == num_set.end()) {
            int currentNum = num;
            int currentStreak = 1;

            // 向上寻找连续的数字
            while (num_set.find(currentNum + 1) != num_set.end()) {
                currentNum += 1;
                currentStreak += 1;
            }

            // 更新最长连续序列的长度
            longest = max(longest, currentStreak);
        }
    }

    return longest;
}

int main() {
    vector<int> nums = {100, 4, 200, 1, 3, 2};
    cout << "最长连续序列的长度: " << longestConsecutive(nums) << endl;
    return 0;
}

File: Python/2-Medium/00128._最长连续序列/00128._最长连续序列.py
from typing import List

def longest_consecutive(nums: List[int]) -> int:
    """
    寻找最长连续序列的长度。
    
    Args:
    nums: List[int] - 一个整数数组
    
    Returns:
    int - 数组中数字连续序列的最长长度
    
    示例:
    >>> longest_consecutive([100, 4, 200, 1, 3, 2])
    4
    """
    # 使用集合去重并提高查找效率
    num_set = set(nums)
    max_length = 0

    for num in num_set:
        # 如果当前数字的前一个数字不在集合中，说明可以从当前数字开始尝试构建连续序列
        if num - 1 not in num_set:
            current_num = num
            current_length = 1
            
            # 向当前数字的右侧查找连续的数字
            while current_num + 1 in num_set:
                current_num += 1
                current_length += 1
                
            # 更新最长连续序列的长度
            max_length = max(max_length, current_length)
    
    return max_length

# 测试代码
test_nums = [100, 4, 200, 1, 3, 2]
print(longest_consecutive(test_nums))